sdks/java/harness/src/main/java/org/apache/beam/fn/harness/HandlesSplits.java

```diff
@@ -35,6 +36,12 @@
   /** Returns the current progress of the active element as a fraction between 0.0 and 1.0. */
   double getProgress();
 
+  String getPtranformId();
```

Comment:
I was under the impression that we would be able to pass this information forward locally through the method without needing to expose it within HandleSplits.

Reply:
Yes, we can pass these information locally for current transform. But I also need these information from `process transform` when current transform is `truncate`.  Whether this change is needed depends on the decision on which transform id and main input id we should use for split from `truncate`. As I mentioned in previous comment:
I think we should still use transform id and main input id from process sized elements. There are 3 possible cases:

- We only have splits on window boundary

- We only have element split on the last window

- We have both element splits and window splits.

If we choose transfrom id and input id from truncate for element split, it's also wrong conceptually. I prefer process sized elements because it will simplify the implementation on the runner side. For example, the runner doesn't need to distinguish window splits or element splits. If we treat truncate and process as a whole, it also make sense to use transform id from process.

Reply:
I believe we should be using the transform ids from the local transform. In the window observing truncate case where we have both element splits from `process` and window splits from `truncate` then:
* the element splits should use the transform/input id from `process`
* the whole window splits should use the transform/input id from `truncate`

We may need changes on Dataflow runner v2 to make sure it supports this.

Reply:
Discussed offline. We decided that window splits are populated with truncate transform id and element splits are populated with process transform id.